tests/unit/compiler/venom/test_common_subexpression_elimination.py

```diff
@@ -0,0 +1,535 @@
+import pytest
+
+from tests.venom_utils import assert_ctx_eq, parse_from_basic_block, parse_venom
+from vyper.venom.analysis.analysis import IRAnalysesCache
+from vyper.venom.passes.common_subexpression_elimination import CSE
+
+
+def _check_pre_post(pre: str, post: str):
```

Comment:
style thing: maybe instead of both `_check_pre_post()` and `_check_pre_post_fn()`, just have `check_pre_post(pre, post, scope="basicblock")`, and have a decision for which parsing function to use:
```
if scope == "basicblock":
    parse = parse_from_basic_block
else:
    assert scope == "function"
    parse = parse_venom
```